src/common/file_stream.h

```diff
@@ -15,82 +15,119 @@
 #include "3rd_party/exception.h"
 #include "common/logging.h"
 
-#ifdef _WIN32
+namespace io = boost::iostreams;
+
+#ifdef _MSC_VER
+
+#include <fcntl.h>
 #include <io.h>
-#include <sys/stat.h>
-#include <sys/types.h>
-#endif
+#include <stdlib.h>
 
-namespace io = boost::iostreams;
+#endif
 
 class TemporaryFile {
 private:
-  int fd_;
+	int fd_;
   bool unlink_;
-  std::string name_;
+	std::string name_;
 
+#ifndef _MSC_VER
   int mkstemp_and_unlink(char* tmpl) {
-#ifdef _WIN32
-    ABORT_IF(true, "mkstemp not available in Windows");
-    int ret = -1;
-#else
     int ret = mkstemp(tmpl);
-#endif
     if(unlink_ && ret != -1) {
       ABORT_IF(unlink(tmpl), "Error while deleting '{}'", tmpl);
     }
     return ret;
   }
+#endif
 
-  int MakeTemp(const std::string& base) {
-    std::string name(base);
-    name += "marian.XXXXXX";
-    name.push_back(0);
-    int ret;
-    ABORT_IF(-1 == (ret = mkstemp_and_unlink(&name[0])),
-             "Error while making a temporary based on '{}'",
-             base);
-    name_ = name;
-    return ret;
-  }
 
-  void NormalizeTempPrefix(std::string& base) {
-    if(base.empty())
-      return;
-    if(base[base.size() - 1] == '/')
-      return;
-    struct stat sb;
-    // It's fine for it to not exist.
-    if(-1 == stat(base.c_str(), &sb))
-      return;
-#ifdef _WIN32
-#define S_ISDIR(m) (((m)&S_IFMT) == S_IFDIR)  // TODO: unify this
+	int MakeTemp(const std::string& base) {
+#ifdef _MSC_VER
+    char* name = _tempnam(base.c_str(), "marian.");
+    ABORT_IF(name == NULL,
+			"Error while making a temporary based on '{}'",
+      base);
+
+    int oflag = _O_RDWR | _O_CREAT | _O_EXCL;
+    if (unlink_) oflag |= _O_TEMPORARY;
+
+		int ret = _open(name, oflag, _S_IREAD | _S_IWRITE);
```

Comment:
I dislike the _ prefix, and that VS forces them upon us. These are the standard Linux functions, there is no value in renaming them on Windows. In my view, we should insist to use the Linux names, and disable the warning explicitly. My version already does that. This is one of the few cases where disabling a warning is more than justified. (The other warning about the so-called "safe" CRT functions like `fopen_s()` which we should not use here.)

Once you change that, I also think you can reconcile at least some of these diverging code paths (e.g. I saw a `_close()` call below that was identical to a `close()` in the Linux version)

Reply:
I also disabled these 2 warnings. I'll take a look at all those changes.